inc/user.class.php

```diff
@@ -4302,13 +4323,22 @@ public function showForgetPassword($email) {
     * @return boolean true if success
     */
    public function forgetPassword($email) {
-      if ($this->getFromDBbyEmail($email,
-                                  "`glpi_users`.`is_active`
-                                   AND NOT `glpi_users`.`is_deleted`
-                                   AND (`glpi_users`.`begin_date` IS NULL
-                                        OR `glpi_users`.`begin_date` < NOW())
-                                   AND (`glpi_users`.`end_date` IS NULL
-                                        OR `glpi_users`.`end_date` > NOW())")) {
+      $condition = [
+         'glpi_users.is_active'  => 1,
+         'glpi_users.is_deleted' => 0,
+         'OR' => [
+            ['glpi_users.begin_date' => null],
+            ['glpi_users.begin_date' => ['<', new QueryExpression('NOW()')]]
+         ],
+         'AND' => [
```

Comment:
this parent AND seems superflu

Reply:
But this is not... Without this one, we have 2 "OR" key in the array, and only the second one stays.